connectivity/FEATURE_BLE/source/common/ble_trace_helpers.h

```diff
@@ -44,11 +44,38 @@ static inline const char* tr_as_array(T item)
     return (mbed_trace_array)((const uint8_t*)&item, sizeof(item));
 }
 
-static inline const char* to_string(UUID uuid)
-{
+static inline const char* to_string(const UUID &uuid) {
     return (mbed_trace_array)(uuid.getBaseUUID(), uuid.getLen());
 }
 
+static inline char* trace_uuid(const UUID& uuid)
+{
+    /* This is bad for two reasons:
+     * 1. It prints the UUID with colons separating the bytes
+     * 2. The bytes will appear in reverse because the array uses little endian storage
+     */
+    // return (char *)(mbed_trace_array)(uuid.getBaseUUID(), uuid.getLen());
+
+    const uint8_t* buf = uuid.getBaseUUID();
+    const uint8_t  len = uuid.getLen();
+
+    const char *hex = "0123456789ABCDEF";
+
+    char* str = new char[len + 1]();
```

Comment:
who releases this?

Reply:
I suggest you use tr_array for now and then make a new PR that allows tr_array to take an optional parameter that allows printing backwards and without colons

Reply:
Okay. Thanks for your suggestion @paul-szczepanek-arm. Thoughts @pan- ?

Reply:
Resolved by 8f28da36be2c0da2b2abfb5ba2120a509d7bb5cc. Now, the function does not use dynamic memory allocation, so there's no longer a memory leak.